packages/core/src/RenderingEngine/StackViewport.ts

```diff
@@ -3046,14 +3046,10 @@ class StackViewport extends Viewport {
 
     if (!imageURI) {
       // Remove the dataLoader scheme since that can change
-      imageURI = imageIdToURI(currentImageId);
-    }
-
-    const referencedImageURI = imageIdToURI(referencedImageId);
-
-    const endsWith = referencedImageId?.endsWith(imageURI);
-    if (endsWith) {
-      return endsWith;
+      imageURI = imageIdToURI(referencedImageId);
+      return true;
```

Comment:
why return true here? I don't understand

Reply:
just removes a redundant check

before when it looked like:
```
if (!imageURI) {
    imageURI = imageIdToURI(referencedImageId);
}

if (imageIdToURI(referencedImageId) === imageURI) {
    return true;
}
```

if the first `if (!imageURI)` was true then the second would always be true, so simplifying it to
```
if (!imageURI) {
    imageURI = imageIdToURI(referencedImageId);
    return true;
} else if (imageIdToURI(referencedImageId) === imageURI) {
    return true;
}
```

saves a call to `imageIdToURI(referencedImageId)` and an extra equivalency check

Reply:
can you do this instead 

```js
const { imageURI } = options;
  
  // Direct URI match check
  const currentImageURI = imageIdToURI(referencedImageId);
  if (!imageURI || currentImageURI === imageURI) {
    return true;
  }
```

Reply:
updated to use that instead